### ISSUE TITLE
Zone.Identifier Files when copying from Windows to WSL filestructure

### ISSUE THREAD

**Issue by Kai-Richardson** [microsoft/WSL#4609]
* Windows build number:  Microsoft Windows [Version 10.0.18990.1]

`uname -a` produces: Linux 4.19.67-microsoft-standard #1 SMP Sun Aug 18 13:37:54 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

* What you're doing and what's happening:  This occurs when using Windows Explorer.
1.  Download a file from an online resource into a windows folder. For example, [this image](https://i.imgur.com/7uEP6ds.png) downloaded into your Downloads folder.
2. Drag/Copy/Move said file into a WSL directory.
3. Notice the file that is also created, named like "movedfile•Zone.Identifier", but instead of a bullet it is the character 0xF03A (Unicode: U+F03A). It does render like a bullet in Windows Explorer though.

* What's wrong / what should be happening instead: The filename•Zone.Identifier NFTS files should not be created.

Contents of the Zone Identifier file are as to be expected, my sample having the contents:
```
[ZoneTransfer]
ZoneId=3
ReferrerUrl=https://i.imgur.com/7uEP6ds.png
HostUrl=https://i.imgur.com/7uEP6ds.png
```

Image:
![image](https://user-images.githubusercontent.com/49656058/67179151-4d558f00-f38a-11e9-9708-ea2ac2b9edb1.png)

**Comment by therealkenc** [comment]
Can't repro here. 

![image](https://user-images.githubusercontent.com/7049700/67180567-d969b580-f38e-11e9-972e-30f5f139a39e.png)


This is on 19002; whether the small version bump makes a difference I doubt. [Zone.Identifier](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-fscc/6e3f7352-d11c-4d76-8c39-2516a9df36e8) does get a quick search hit. Said [functionality](https://docs.microsoft.com/en-us/previous-versions/windows/internet-explorer/ie-developer/platform-apis/ms537183(v=vs.85)?redirectedfrom=MSDN) is news to me.

On a lark, from your WSL2 bash prompt, what is the output of:
```
$ powershell.exe "Get-CimInstance -Namespace root/SecurityCenter2 -ClassName AntivirusProduct" | grep displayName
```

**Comment by 0xbadfca11** [comment]
I can reproduce with both WSL1 and WSL2 at 19002.
Zone.Identifier may not exist for various reasons, such as the ["Do not save zone information in attachments" policy](https://gpsearch.azurewebsites.net/#26) or deleted by Windows Smart Screen when opening a file.
![image](https://user-images.githubusercontent.com/7390559/67210281-c79e1600-f453-11e9-8dda-8285bfb76356.png)

**Comment by Kai-Richardson** [comment]
@therealkenc `displayName              : Windows Defender`

**Comment by Kai-Richardson** [comment]
@0xbadfca11 Regarding your link to the azure windows documentation, I have no Attachments key in `HKCU\Software\Microsoft\Windows\CurrentVersion\Policies\Attachments`. Therefore, if the documentation is correct, `If you do not configure this policy setting, Windows marks file attachments with their zone information. `

**Comment by therealkenc** [comment]
> If you do not configure this policy setting, Windows marks file attachments with their zone information.

I didn't configure anything "special" in my policy (almost never do as a rule) but copying the file didn't produce the artifact. It does for you and 0xbadfca11.

I suppose the experiment to try is drag a file out of Downloads and onto some Windows share on some other computer on your local network (or a VM). If that Zone.Identifier file also appears in a "normal" network share scenario then this smells by-design-Windows.

I think (?) the variable here might be in your step (1) "Download a file from an online resource into a windows folder". You didn't say how. I used Chrome, but it could have been [`curl`](https://devblogs.microsoft.com/commandline/tar-and-curl-come-to-windows/). The fact that file came from `https://i.imgur.com/` was long lost to the wind by the time my drag-and-drop happened.

**Comment by Kai-Richardson** [comment]
@therealkenc Here's the steps I take exactly, using Firefox (reproduced on Chrome):

1. Navigate to [an image url](https://i.imgur.com/7uEP6ds.png).
2. Right click and Save Image As
3. Choose location on my Windows NTFS C: Drive, such as `C:\Users\zekai\Desktop`
4. Drag (Click and hold) or copy/paste (CTRL+C & CTRL+P) PNG file to a WSL filespace, such as `\\wsl$\Ubuntu\home\cassini\github`
5. Nothing appears at first, but click the refresh button on explorer.exe in the top right.
6. Notice new Zone.Identifier file.

Here's my output from `df -Th`, to list filesystem/partition information.
```
df -Th
Filesystem     Type      Size  Used Avail Use% Mounted on
/dev/sdb       ext4      251G  9.0G  230G   4% /
tools          9p        477G  303G  174G  64% /init
none           devtmpfs  7.3G     0  7.3G   0% /dev
none           tmpfs     7.3G  8.0K  7.3G   1% /run
none           tmpfs     7.3G     0  7.3G   0% /run/lock
none           tmpfs     7.3G     0  7.3G   0% /run/shm
none           tmpfs     7.3G     0  7.3G   0% /run/user
tmpfs          tmpfs     7.3G     0  7.3G   0% /sys/fs/cgroup
C:\            9p        477G  303G  174G  64% /mnt/c
D:\            9p        932G  531G  402G  57% /mnt/d
```

**Comment by therealkenc** [comment]
> Notice new Zone.Identifier file.

Thanks for the detailed steps. Caught it; same repro,

![image](https://user-images.githubusercontent.com/7049700/67615590-17961900-f783-11e9-8d8c-bab975233c56.png)

Below seems to be a hint. "This file came from another computer and might be blocked to help protect this computer." 

![image](https://user-images.githubusercontent.com/7049700/67615606-5e840e80-f783-11e9-9953-a03239a0a494.png)

Clicking that "unblock" button cures. Which puts this (rather unfortunately) in by-design territory, and worse, out-of-scope WSL territory.

**Comment by therealkenc** [comment]
> Notice new Zone.Identifier file.

Thanks for the detailed steps. Caught it; same repro,

![image](https://user-images.githubusercontent.com/7049700/67615738-b40ceb00-f784-11e9-963f-e8a459b1e7c1.png)


Below seems to be a hint. "This file came from another computer and might be blocked to help protect this computer." 


![image](https://user-images.githubusercontent.com/7049700/67615606-5e840e80-f783-11e9-9953-a03239a0a494.png)

Clicking that "unblock" button cures. Which puts this (rather unfortunately) in by-design territory, and worse, out-of-scope WSL territory.

**Comment by Kai-Richardson** [comment]
@therealkenc 
> Clicking that "unblock" button cures. Which puts this (rather unfortunately) in by-design territory, and worse, out-of-scope WSL territory.

I would argue that having extraneous files pop into existence every time something downloaded is copied over to the WSL filesystem should be classified as a bug and not by-design.

Since these .Identifiers are quite useless once they get moved WSL, why not simply ignore the NTFS alternate file stream data during the copy process between file systems. This issue definitely has caused me a bit of pain when I download zipped directories and bring them over to WSL, leading to a proliferation of dozens of .Identifier files.

This command (PowerShell) lists all files with an associated Identifier stream in a directory:
`Get-ChildItem -Recurse | Get-Item -Stream Zone.Identifier -ErrorAction SilentlyContinue | Select-Object FileName`

This is how you remove them with PowerShell:
`Remove-Item .\example.exe -Stream Zone.Identifier`

If PowerShell has the tools to remove these stream identifiers, why can't WSL remove them upon file copy over?

**Comment by therealkenc** [comment]
>  why not simply ignore the NTFS alternate file stream data during the copy process between file systems

Likely because WSL doesn't have the faintest idea whether (scare quote) "those files" should be copied or not. Also because axiom that if the word "simply" is evoked, the premise is near always incorrect.

> I would argue that having extraneous files pop into existence every time something downloaded is copied over to the WSL filesystem should be classified as a bug and not by-design.

Me too; although _someone_ on the _some_ team somewhere at MSFT thought "This file came from another computer and might be blocked" was a good idea.

This all said, I did the experiment of copying the untrused (?) file onto a local share, and there isn't any "•Zone.Identifier" shenanigans. 

![image](https://user-images.githubusercontent.com/7049700/67732705-7349e780-f9b9-11e9-9e28-f8653579fccf.png)

This issues hasn't been closed by-design. Someone over my pay grade would need to do that. If it were, you'd already know.

**Comment by riverar** [comment]
Explorer is attempting to set the NTFS ADS and WSL is not handling this correctly. The 0x003A (colon) getting altered to 0xF03A (dot) sounds like text encoding or sanitation gone awry.

Stack from `18363.418`:
```0	FLTMGR.SYS	FltpPerformPreCallbacks + 0x2fc	0xfffff802213e3a5c	C:\WINDOWS\System32\drivers\FLTMGR.SYS
1	FLTMGR.SYS	FltpPassThroughInternal + 0x90	0xfffff802213e35a0	C:\WINDOWS\System32\drivers\FLTMGR.SYS
2	FLTMGR.SYS	FltpCreate + 0x2f3	0xfffff8022141bd13	C:\WINDOWS\System32\drivers\FLTMGR.SYS
3	ntoskrnl.exe	IofCallDriver + 0x59	0xfffff80223831f39	C:\WINDOWS\system32\ntoskrnl.exe
4	ntoskrnl.exe	IoCallDriverWithTracing + 0x34	0xfffff80223830fe4	C:\WINDOWS\system32\ntoskrnl.exe
5	ntoskrnl.exe	IopParseDevice + 0x62b	0xfffff80223de5ffb	C:\WINDOWS\system32\ntoskrnl.exe
6	ntoskrnl.exe	IopParseFile + 0xc7	0xfffff80223ebccc7	C:\WINDOWS\system32\ntoskrnl.exe
7	ntoskrnl.exe	ObpLookupObjectName + 0x78f	0xfffff80223decfcf	C:\WINDOWS\system32\ntoskrnl.exe
8	ntoskrnl.exe	ObOpenObjectByNameEx + 0x201	0xfffff80223deb431	C:\WINDOWS\system32\ntoskrnl.exe
9	ntoskrnl.exe	IopCreateFile + 0x820	0xfffff80223e30300	C:\WINDOWS\system32\ntoskrnl.exe
10	ntoskrnl.exe	IoCreateFileEx + 0x11d	0xfffff80223e2f9ad	C:\WINDOWS\system32\ntoskrnl.exe
11	LXCORE.SYS	LxpUtilOpenFileEx + 0x356	0xfffff80226752d6a	C:\WINDOWS\system32\drivers\LXCORE.SYS
12	LXCORE.SYS	LxpUtilOpenFile + 0x45	0xfffff80226752a09	C:\WINDOWS\system32\drivers\LXCORE.SYS
13	LXCORE.SYS	LxVolFsInodeOpen + 0xa0	0xfffff802267739e0	C:\WINDOWS\system32\drivers\LXCORE.SYS
14	LXCORE.SYS	VfsInodeOpen + 0x176	0xfffff8022675f9ae	C:\WINDOWS\system32\drivers\LXCORE.SYS
15	LXCORE.SYS	VfsOpenInodeEx + 0x1bc	0xfffff8022676493c	C:\WINDOWS\system32\drivers\LXCORE.SYS
16	LXCORE.SYS	VfsOpenFile + 0xa3	0xfffff80226764723	C:\WINDOWS\system32\drivers\LXCORE.SYS
17	LXCORE.SYS	LxpUtilOpenUserPathAt + 0xab	0xfffff80226752ec3	C:\WINDOWS\system32\drivers\LXCORE.SYS
18	LXCORE.SYS	LxpOpenHelper + 0x90	0xfffff80226738528	C:\WINDOWS\system32\drivers\LXCORE.SYS
19	LXCORE.SYS	LxpSyscall_OPENAT + 0x9	0xfffff8022673faa9	C:\WINDOWS\system32\drivers\LXCORE.SYS
20	LXCORE.SYS	LxpSysDispatch + 0x114	0xfffff80226739944	C:\WINDOWS\system32\drivers\LXCORE.SYS
21	LXCORE.SYS	PicoSystemCallDispatch + 0x11	0xfffff80226756df1	C:\WINDOWS\system32\drivers\LXCORE.SYS
22	ntoskrnl.exe	PsPicoSystemCallDispatch + 0x1f	0xfffff802240cbb5f	C:\WINDOWS\system32\ntoskrnl.exe
23	ntoskrnl.exe	KiSystemServiceUser + 0x76	0xfffff802239d28f8	C:\WINDOWS\system32\ntoskrnl.exe
```

**Comment by thisguychris** [comment]
@therealkenc I've been getting this for months too, and I think I know why you are not getting any the Zone Identifier file. If I copy and paste to a WSL2 folder, I won't see the identifier file yet. But if I go a folder up, then go back again to the original folder, I now see the identifier file.

here's the [video ](https://monosnap.com/file/ov1aveYCqoLC0UCO3WGDiiuYXlLFuP) of what I'm seeing.

I'm using WSL2, and here's my build number.
Microsoft Windows [Version 10.0.19037.1]

**Comment by Stanzilla** [comment]
Yeah this is happening here as well, pretty annoying.

**Comment by ThaDaVos** [comment]
I hope this gets fixed soon as it's really annoying when, like mentioned already, you copy a extracted zip file to the WSL directory

**Comment by thisguychris** [comment]
@therealkenc hey there, 👋🏻. I was wondering if you are now seeing this a bug after I posted [details on how to reproduce it to give you more context](https://github.com/microsoft/WSL/issues/4609#issuecomment-547229173)?

If you can reproduce it, any chance we can label this as a bug?

**Comment by riverar** [comment]
@therealkenc Please label as bug.

**Comment by therealkenc** [comment]
Precedent (for better or worse) is the dev responsible makes that call.  My # 1529 was opened Dec 2016 and I only took the rare liberty of marking it as a bug in May 2018 (a regretful precedent). A personal favorite # 2779 managed to make it from Dec 2017 to fixedinisiders Sept 2019 with no bug tag at all (noting # 2779 was also a terrible UX).

But in the interest of popular demand, sure. Noting the significance of the already-existing open tag status (and lack of by-design tag) on this issue is probably being underestimated, and the significance of the newly minted bug tag is probably being overestimated.

**Comment by riverar** [comment]
@therealkenc Thanks! It helps externals a lot when it comes to badgering the folks responsible and shows up in quantitative reporting.

**Comment by thisguychris** [comment]
Hey, @therealkenc thanks for adding the label. 👋🏻 I'm actually more interested if it's reproducible in your end, that way, everyone in this thread would be in agreement it is happening. It somehow giving a mixed signal that the bug is inconsistent. If it's consistent then it is going to be easier to debug.

Tagging, @craigloewen-msft, and @benhillis, **_maybe_** this is a low hanging fruit that could easily be squashed, [as described by a post of @riverar](https://github.com/microsoft/WSL/issues/4609#issuecomment-549271706)

**Comment by Kai-Richardson** [comment]
@thisguychris they reproduced it in this reply: https://github.com/microsoft/WSL/issues/4609#issuecomment-546577426

**Comment by thisguychris** [comment]
@Kai-Richardson my bad, I totally missed that, 🙏🏻 sorry about that @therealkenc

**Comment by staticle** [comment]
A Simple Solution just run **`cd ~ && find . -name "*:Zone.Identifier" -type f -delete**`

**Comment by staticle** [comment]
> @therealkenc I've been getting this for months too, and I think I know why you are not getting any Zone Identifier file. If I copy and paste to a WSL2 folder, I won't see the identifier file yet. But if I go a folder up, then go back again to the original folder, I now see the identifier file.
> 
> here's the [video ](https://monosnap.com/file/ov1aveYCqoLC0UCO3WGDiiuYXlLFuP) of what I'm seeing.
> 
> I also want to add that I copied from a `Downloads` folder of Windows to a folder of my `\\wsl$\ubuntu...\home\username\.local\_temp`
> 
> I'm using WSL2, and here's my build number.
> Microsoft Windows [Version 10.0.19037.1]

A Simple Solution just run cd ~ && find . -name "*:Zone.Identifier" -type f -delete

**Comment by staticle** [comment]
> * Windows build number:  Microsoft Windows [Version 10.0.18990.1]
> 
> `uname -a` produces: Linux Mariner 4.19.67-microsoft-standard #1 SMP Sun Aug 18 13:37:54 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
> 
> * What you're doing and what's happening:  This occurs when using Windows Explorer.
> 
> 1. Download a file from an online resource into a windows folder. For example, [this image](https://i.imgur.com/7uEP6ds.png) downloaded into your Downloads folder.
> 2. Drag/Copy/Move said file into a WSL directory.
> 3. Notice the file that is also created, named like "movedfile•Zone.Identifier", but instead of a bullet it is the character 0xF03A (Unicode: U+F03A). It does render like a bullet in Windows Explorer though.
> 
> * What's wrong / what should be happening instead: The filename•Zone.Identifier NFTS files should not be created.
> 
> Contents of the Zone Identifier file are as to be expected, my sample having the contents:
> 
> ```
> [ZoneTransfer]
> ZoneId=3
> ReferrerUrl=https://i.imgur.com/7uEP6ds.png
> HostUrl=https://i.imgur.com/7uEP6ds.png
> ```
> 
> Image:
> ![image](https://user-images.githubusercontent.com/49656058/67179151-4d558f00-f38a-11e9-9708-ea2ac2b9edb1.png)

A Simple Solution just run cd ~ && find . -name "*:Zone.Identifier" -type f -delete

**Comment by eriksenonline** [comment]
I will not say this is a simple solution. It is a workaround for somthing we should not have in our linux files.

**Comment by NiklasBr** [comment]
An XML file got this identifier file:
```
[ZoneTransfer]
ZoneId=3
```

```bash
niklas@Fractal:~$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.4 LTS
Release:        18.04
Codename:       bionic
niklas@Fractal:~$ uname -a
Linux Fractal 4.19.104-microsoft-standard #1 SMP Wed Feb 19 06:37:35 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by riverar** [comment]
Still a bug in Version 2004 GA, 19041.264.

**Comment by sanderkooger** [comment]
Having the same issue here. Help is welcome

**Comment by drewstaylor** [comment]
> Can't repro here.
> 
> ![image](https://user-images.githubusercontent.com/7049700/67180567-d969b580-f38e-11e9-972e-30f5f139a39e.png)
> 
> This is on 19002; whether the small version bump makes a difference I doubt. [Zone.Identifier](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-fscc/6e3f7352-d11c-4d76-8c39-2516a9df36e8) does get a quick search hit. Said [functionality](https://docs.microsoft.com/en-us/previous-versions/windows/internet-explorer/ie-developer/platform-apis/ms537183(v=vs.85)?redirectedfrom=MSDN) is news to me.
> 
> On a lark, from your WSL2 bash prompt, what is the output of:
> 
> ```
> $ powershell.exe "Get-CimInstance -Namespace root/SecurityCenter2 -ClassName AntivirusProduct" | grep displayName
> ```

It's likely you just don't see them in file explorer, because they will only display when you have file explorer set to show hidden files. Also, I have so far only experienced this issue in WSL 2.

**Comment by riverar** [comment]
@drewstaylor  It's already been confirmed as reproducible, you're responding to an old post.

**Comment by benhillis** [comment]
@SvenGroot - Can you please take a look?  Looks like we should be handling these alternate data streamd differently.

**Comment by TasinAhmed** [comment]
Any update on this? This is very annoying.

**Comment by nhart-alf** [comment]
Seeing this on 19041.508 GA, WSL2, Ubuntu.
VScode running from WSL shows the files as they turn up and I can remove them from there but it's still irritating.
Repro steps:

1. download a file via windows
2. copy file to \\wsl$\Ubuntu\somepath
3. ADS files are now present within the linux filesystem

**Comment by antoineheseque** [comment]
Same "bug" on both computer with WSL2 and the last Ubuntu version. Exactly the same steps (follow the tutorial on the website, and copy a file from Windows to \\wsl$\Ubuntu\home\...

**Comment by leighfwarren** [comment]
Same bug here, it's really annoying and wrong.

**Comment by frumbert** [comment]
I also get `:com.dropbox.attrs` as well as `:Zone.Identifer` files everwhere. Plays havoc with git repos when I copy those into Ubuntu!!

Extended attrbiutes / alternate ntfs file streams or whatever they are should NOT get copied.

**Comment by toknsi** [comment]
the cause of  `Zone.Identifier` is `Windows marks file attachments with information about their zone of origin`
```powershell
get-item .\smile.png -stream *
PSChildName   : smile.png:Zone.Identifier
FileName      : C:\workspace\programming--windows\smile.png
Stream        : Zone.Identifier
Length        : 153
```
disable `Zone.Identifier`

https://getadmx.com/?Category=Windows_10_2016&Policy=Microsoft.Policies.AttachmentManager::AM_MarkZoneOnSavedAtttachments

**Comment by Kai-Richardson** [comment]
> the cause of `Zone.Identifier` is `Windows marks file attachments with information about their zone of origin`

We've known the actual cause for almost a year now. See riverar's comment.

> Explorer is attempting to set the NTFS ADS and WSL is not handling this correctly. The 0x003A (colon) getting altered to 0xF03A (dot) sounds like text encoding or path sanitation gone awry.
https://github.com/microsoft/WSL/issues/4609#issuecomment-549271706

**Comment by riverar** [comment]
@therealkenc Hey Ken, can you help drum up internal noise about this year long issue now?

cc: @SvenGroot @clintrutkas @bitcrazed

**Comment by ertomas** [comment]
I just got this bug. 

I saved an email attachment from Outlook directly to WSL via \\wsl on explorer and got the Zone.Identifier file.

**Comment by todpale** [comment]
Got this bug too.

**Comment by MattGarnettWelsh** [comment]
All these little strange behaviours from Microsoft in WSL make it really hard to fully embrace it! As mentioned - this issue is continuously playing havoc with git repos and general file management.

**Comment by bitcrazed** [comment]
Hey @MattGarnettWelsh - appreciate the feedback.

Alas, our team is currently swamped with high-pri work, but are aware of the issue and will respond as soon as they come up for air.

To make sure we're not missing anything unexpected, are there other issues you're seeing? If so, please be sure to find/file issues accordingly.

Thanks for your patience.

**Comment by Neutrino-Sunset** [comment]
I get this issue with js files created locally, nothing to do with downloading anything.

**Comment by bitcrazed** [comment]
Hey @Neutrino-Sunset - could you share a repro?

**Comment by riverar** [comment]
@bitcrazed There's quite a few repros in this thread, come on.

**Comment by leighfwarren** [comment]
It happens every time I copy a file in, it shouldn't be too hard.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by Neutrino-Sunset** [comment]
How can I provide a repo? I installed the Ubuntu distro, set it to WSL 2, whenever I copy a file into Ubuntu/home in the WSL filesystem it using Windows Explorer it creates Zone.Identifier files. Not sure how I can create a repo of my file system.

**Comment by bitcrazed** [comment]
@riverar - So you can create a JavaScript file locally in Windows, and copy it to a WSL folder via \\wsl$... and you see ghost files?

![image](https://user-images.githubusercontent.com/961950/98581863-f5304a00-2276-11eb-9143-87977cdbbc8c.png)

**Comment by riverar** [comment]
@bitcrazed The file needs an NTFS alternate data stream. The user is probably unaware their file contains NTFS ADS of some sort.

*Example 1:*
```ps
"[ZoneTransfer]\nZoneId=3" | Set-Content -Path test.js -Stream Zone.Identifier
```

*Example 2:*
```ps
"hello world" | Set-Content -Path test2.js -Stream BleepBloop
```

**Comment by Kai-Richardson** [comment]
Here's my repo that I shared at the top of this thread, over a year ago: https://github.com/microsoft/WSL/issues/4609#issuecomment-546574922
Here's it being successfully replicated: https://github.com/microsoft/WSL/issues/4609#issuecomment-546577426
Those were due to NTFS ADS.

According to [this comment](https://github.com/microsoft/WSL/issues/4609#issuecomment-712665398), this bug can also be replicated with NTFS extended attributes.

**Comment by leighfwarren** [comment]
This happens when you copy it using Windows explorer

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by riverar** [comment]
![image](https://user-images.githubusercontent.com/475132/98583877-ee570680-2279-11eb-9fcc-366a15ab7ccf.png)

Funny enough, at least in Windows 10 20H2, this dialog appears in some cases. While it states it will copy the file without NTFS ADS, it still copies all NTFS ADS.

**Comment by bitcrazed** [comment]
@riverar I am very well aware that the issue as described  requires an ADS, which is why I was surprised when  @Neutrino-Sunset [stated](https://github.com/microsoft/WSL/issues/4609#issuecomment-723640387) that they "get this issue with js files created locally, nothing to do with downloading anything". 

Such files should not contain any ADS' and thus should not exhibit the behavior. Hence [my request](https://github.com/microsoft/WSL/issues/4609#issuecomment-724180737) **_to them_**, for a repro/example of _**their**_ issue.

As per my screenshot above, I have tried to repro the case of a locally created .js file which I copied into my Ubuntu distro's filesystem from Windows (I also tried doing so via File Explorer) ... but was unable to repro **_their_** case as described.

**Comment by riverar** [comment]
@bitcrazed Ok.

**Comment by TasinAhmed** [comment]
Any update on this? This is the only issue holding me back from going back to WSL2

**Comment by riverar** [comment]
@TasinAhmed Is it really holding you back though?

**Comment by bitcrazed** [comment]
@Kai-Richardson @TasinAhmed et al:

The team are exploring this issue. 

However, there is a work around for this issue: Unblock files downloaded download from the public internet to your PC to remove [the mark of the web](https://www.howtogeek.com/70012/what-causes-the-file-downloaded-from-the-internet-warning-and-how-can-i-easily-remove-it/), __before trying to copy said files into WSL__.

⚠ __WARNING: ONLY UNBLOCK FILES THAT YOU HAVE VERIFIED AND CAN BE TRUSTED!__ ⚠

Below, for example, is a screenshot of two File Explorers (top) and two terminal panes (bottom). On the left is Windows, on the right is Ubuntu in WSL. 

I locally created a simple image using Paint.net and saved it as test.jpg & test.png. 

I also downloaded an image from the public internet which, as can be seen in the PowerShell Terminal-pane (bottom left) is the only file in the folder which is branded with the 'mark of the web' ADS stream. I then copied this file, opened it's file properties and clicked the checkbox at the bottom of the file properties dialog to "unblock" the file. This removes the 'mark of the web' ADS stream from the copied file (but not the original).

I copied these four files to my Ubuntu home folder by copying and pasting via File Explorer. 

As you can see, only the original image file downloaded from the public internet branded with 'the mark of the web' ADS stream is accompanied by an offending `...Zone.Identifier` file.

![image](https://user-images.githubusercontent.com/961950/100644473-33f27680-32f0-11eb-9233-e7ebf3fa5f39.png)

HTH.

**Comment by GavinRay97** [comment]
Not to be tiresome because I know this thread has been open a while and you folks are looking into it, but this phenomenon is a bit unpleasant

Would there be any possibility of a WSL config option for "automatically delete ZoneIdentifier files when copying between hosts"? This is kind of a duct-tape fix, but at least wouldn't be an issue anymore.

Thank you =)

**Comment by omarbadran** [comment]
I think this has been solved with the latest windows update. I copied some files today and did not encounter the problem, which I used to every time I copy any file from Windows to WSL. I'm on the insider preview version I'm not sure if this has reached the mainline versions yet.

**Comment by therealkenc** [comment]
Thanks Omar. I'll second, based on the [repro](https://github.com/microsoft/WSL/issues/4609#issuecomment-546577426) from October. /fixed 20279 (or thereabouts)

![image](https://user-images.githubusercontent.com/7049700/103423370-0e5cc500-4b5b-11eb-8d38-75caeac2cdbf.png)

**Comment by GavinRay97** [comment]
> I think this might have been resolved with the latest update. I copied some files today and did not encounter the problem, which I used to every time I copied any file from Windows to WSL. I'm on the insider preview version so I'm not sure if this has reached the mainline version yet.

Ahh okay @omarbadran. I'm on Insiders, though maybe I need to update

![image](https://user-images.githubusercontent.com/26604994/103423563-3ce79980-4b75-11eb-8a85-85c5dd430945.png)

**Comment by wespiard** [comment]
I have this problem when moving files from Windows to WSL via Windows Explorer GUI. If I use the "mv" command from within my Ubuntu shell, e.g., `mv /mnt/c/<path_to_windows_file> /<path_to_WSL_directory>` then I don't get the ZoneIdentifier files. 

By doing this, I don't have to change security options. 

Not sure if this helps or not, but thought I'd mention it.

**Comment by shyney7** [comment]
Im still getting this issue.
Im on Windows:
Edition	Windows 10 Education
Version	20H2
Installiert am	‎28.‎05.‎2020
Betriebssystembuild	19042.804
Windows Feature Experience Pack 120.2212.551.0
My Ubuntu Linux Kernel is: Linux 5.4.72-microsoft-standard-WSL2

**Comment by ertomas** [comment]
It just happened to me again yesterday. I saved an Outlook attachtment directly to  \\wsl$\distro and got the zone identifier file. Windows 20H2/Ubuntu 20.04  De: MarcelEnviado: jueves, 25 de febrero de 2021 10:07 a. m.Para: microsoft/WSLCC: ertomas; CommentAsunto: Re: [microsoft/WSL] Zone.Identifier Files when copying from Windows to WSL filestructure (#4609) Im still getting this issue.Im on Windows:Edition Windows 10 EducationVersion 20H2Installiert am ‎28.‎05.‎2020Betriebssystembuild 19042.804Windows Feature Experience Pack 120.2212.551.0My Ubuntu Linux Kernel is: Linux 5.4.72-microsoft-standard-WSL2—You are receiving this because you commented.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by TasinAhmed** [comment]
I still have this issue. I am running:

![image](https://user-images.githubusercontent.com/28973106/110368821-8b069680-8017-11eb-855b-ac7218e3e9de.png)

**Comment by j0nnybrav079** [comment]
I can reproduce this error, too. But not every time, if I try it about 5-10 times, it will write at least one Zone.Identifier file

**Comment by Raltrwx** [comment]
Still happening for me

![image](https://user-images.githubusercontent.com/20295023/112697878-1ec8c700-8e4e-11eb-96ce-ea03d252e6e9.png)

Win 10 OS Build 19041.867 & WSL 2

**Comment by NiklasBr** [comment]
@therealkenc this still happens in beta builds, it appears the closing of this bug was premature.

**Comment by domainfun** [comment]
I've just learned about these undesired "...:Zone.Identifier" files when copying/moving files from C:\> to Linux on WSL2 (such as to \\\\wsl$\\...).  Could this issue please be resolved, as this is actually a security issue for us.

It's all too easy to disclose the URL where we may have downloaded a file from.  All it needs is an rsync from WSL to some other (backup) location and these "...:Zone.Identifier" files quickly show up in unexpected places.

Many thanks.

**Comment by sanderkooger** [comment]
You could use ssh to copy files as workarround

**Comment by AnthonyCost** [comment]
I've still had this issue for months, I've been searching everywhere for a solution but hopefully they can patch this soon.

**Comment by Asterovim** [comment]
same problem. Please help us.

**Comment by therealkenc** [comment]
> same problem. Please help us.

If anyone is still experiencing the `Zone.Identifier` problem on 20279 or better, please post the repro steps and screencap similar to [above](https://github.com/microsoft/WSL/issues/4609#issuecomment-753032182) including `cmd.exe /c ver`. 19xxx did not roll in 21h1.

**Comment by phpguru** [comment]
@therealkenc 

> still experiencing on 20279 or better

20279 is a version number of what? And where do I find mine?

**Comment by NiklasBr** [comment]
> 20279 is a version number of what? And where do I find mine?

It's a [Windows Preview build version](https://blogs.windows.com/windows-insider/2020/12/14/announcing-windows-10-insider-preview-build-20279/).

You can find your build version in *Settings: System: About*

**Comment by therealkenc** [comment]
> You can find your build version in Settings: System: About

[....or](https://github.com/microsoft/WSL/issues/4609#issuecomment-829483699) `cmd.exe /c ver`

**Comment by StarpTech** [comment]
Issue still exists:

Repro:

1. Download https://github.com/ogham/exa/releases/download/v0.10.1/exa-linux-x86_64-v0.10.1.zip
2. Open explorer and navigate to `~` on the WSL host.
3. Extract the `/bin/exa` file to a new directory in `~`.
4. `exa` and `exaZone.Identifier` are created.

Version: Microsoft Windows [Version 10.0.19042.985]

**Comment by riverar** [comment]
@StarpTech As @therealkenc implies above, this appears to be fixed in 20279+ (i.e. Insider builds).

**Comment by aravindvnair99** [comment]
I think starting a new issue is better as I don't think closed issues are being monitored.

**Comment by rehanhaider** [comment]
Can confirm this issues still exists on Windows 11 while downloading a file using browser directly onto WSL.

**Comment by NiklasBr** [comment]
@rehanhaider Microsoft (@therealkenc) closed this issue as apparently fixed in Dec 2020, please also add your voice to #7456 which is the follow-up issue.

**Comment by SheepReaper** [comment]
#7456 is the new version of this issue. Please go there and upvote the OP.

**Comment by MattGarnettWelsh** [comment]
Still happening on Ubuntu 20.04 in WSL2. Extracting a zip file to a path mapped to windows adds a zone identifier file to each file:

![image](https://user-images.githubusercontent.com/13387497/158580430-d130ba78-90e5-4c33-b00d-ff40c95419ba.png)

**Comment by Kryuko** [comment]
Issue still exists with latest w10 and wsl2 build.

**Comment by equiman** [comment]
I've enabled the `Do not preserve zone information in file attachments` but it is still creating these annoying `*:ZoneIdentifier` files. While this can be solved. I've created an alias to run after copying files.

```bash
alias rzi="rm -rf **/*Zone.Identifier"
```

Also, I always add `**/*Zone.Identifier` to `.gitignore` files on my project.

**Comment by NiklasBr** [comment]
@equiman see #7456 

This issue was closed despite it not being fixed.

**Comment by JoshuaJarman** [comment]
2022-05-25 Win11 Pro Insiders 25120rs_prerelease.220513-1246

Problem still persists. 😢 such a pita to have to delete 2 files for every 1 file copied.

**Comment by rayfoss** [comment]
> @equiman see #7456
> 
> This issue was closed despite it not being fixed.

Very much so in the latest Windows 11 with a brand new high-end laptop. This issue caused me to send a customer a corrupt zip file that did not validate in their system as the bullet point separator is an invalid character with their file system and the zip fails to extract. This is a **huge** critical regression grade **bug**.

**Comment by OliverKK** [comment]
Helped me to get rid of the files:
`find . -name "*:Zone.Identifier" -type f -delete`

**Comment by rafaelgildin** [comment]
> Helped me to get rid of the files: `find . -name "*:Zone.Identifier" -type f -delete`
That worked for me. Thanks @OliverKK

**Comment by bmlyon** [comment]
Can confirm this is still happening. It would be nice to get this fixed so we don't have to remember to do the workarounds every time before we commit in Git.

**Comment by BarsMonster** [comment]
Still happens in Windows 11 (10.0.22621.674), and WSL2 kernel 5.15.68.1

**Comment by rafaelgildin** [comment]
> Still happens in Windows 11 (10.0.22621.674), and WSL2 kernel 5.15.68.1

Try to update windows 11.
It disappeared to me.
If it doesn't work try the code above

**Comment by NiklasBr** [comment]
We have now celebrated the third birthday of this issue and it still happens! 🥳 🎊

**Comment by SheepReaper** [comment]
I'm not convinced that this is a bug. This is the expected behavior of the ADS when downloading to a Windows share where the filesystem isn't ntfs. You're interacting with the share if you're using windows explorer to download to a wsl path. You don't see this if for example you perform file operations from within wsl or curl, but that's also expected. If you really hate the existence of those files, don't drag and drop through explorer or copy into the share. Alternatively disable the preservation of zone data via local policy. I really don't see M$ making an exception for a "magic" share. The policy you want is User Configuration\Policies\Administrative Templates\Windows Components\Attachment Manager\Do not preserve zone information in file attachments (set it to enabled)

**Comment by riverar** [comment]
@SheepReaper As mentioned in 2019, neither Windows or Linux support `·Zone.Identifier` so seems pointless to generate. [This was fixed in an Insider build](https://github.com/microsoft/WSL/issues/4609#issuecomment-753032182) but appears to have regressed.

@bitcrazed Can you re-open and assist here?

**Comment by SheepReaper** [comment]
Ntfs supports Alternate Data Streams. https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-fscc/6e3f7352-d11c-4d76-8c39-2516a9df36e8 windows supports ADS. Explorer is sending all streams. The share fs (ext4) does not. Rather than losing information, the default behavior is to create the zone identifier file. Between wsl 1 and 2 the way to access your Linux file systems from windows was replaced with a normal Windows share. I don't know that this has ever worked the way you expect in wsl2. This is default behavior for ADS on windows...

**Comment by riverar** [comment]
@SheepReaper I understand NTFS ADS (see posts I linked to), but my point is that after being copied into `\\WSL$`, these `·` prefaced files are created. These files are not useful for any OS, so begs the question why they're created to begin with. It seems Microsoft agreed as they previously fixed the issue.

**Comment by nathanneuro** [comment]
This wasn't happening for me for the first few months of using WSL 2, but then started happening just recently. Pretty annoying. Why was the issue closed if it's still an unfixed problem?

**Comment by NiklasBr** [comment]
@nathanneuro add your voice to https://github.com/microsoft/WSL/issues/7456

**Comment by richardheller** [comment]
Until fixed, just copy the files over using wsl command line the issue does not occur...
![image](https://user-images.githubusercontent.com/94970791/207619641-75892a80-8dcb-4b83-a768-f5101b08755d.png)

**Comment by fonnesbeck** [comment]
This is clearly still an issue. Should I open a new bug report?

**Comment by NiklasBr** [comment]
@fonnesbeck No, there is already a new issue opened: https://github.com/microsoft/WSL/issues/7456

**Comment by tomba4** [comment]
@SvenGroot Can you please explain how come this issue is marked as fixed when in fact it's not?

**Comment by davidruizdiaz** [comment]
I also have the same problem and when trying to delete with the command `find . -name "*:Zone.Identifier" -type f -delete` it gives me the error `find: cannot delete ‘./file:Zone.Identifier’: Read-only file system`
Does anyone know of another way to delete the file?

**Comment by SheepReaper** [comment]
> I also have the same problem and when trying to delete with the command `find . -name "*:Zone.Identifier" -type f -delete` it gives me the error `find: cannot delete ‘./file:Zone.Identifier’: Read-only file system` Does anyone know of another way to delete the file?

Whatever filesystem you're working on appears to be mounted read-only. You won't be able to delete any files on it until you address that.

@tomba4 It's the unfortunate nature of issue handling. Closed issues may as well be locked for all the attention they will get. There's already a new issue opened. Please go there and thumb up the OP to increase the visibility of the open issue: https://github.com/microsoft/WSL/issues/7456

**Comment by Andrewlevvv23** [comment]
> I also have the same problem and when trying to delete with the command `find . -name "*:Zone.Identifier" -type f -delete` it gives me the error `find: cannot delete ‘./file:Zone.Identifier’: Read-only file system` Does anyone know of another way to delete the file?

You don't have the correct command line, you need it like this:  find . -type f -name "*.Identifier" -exec rm -f {} \;

**Comment by darylknight** [comment]
I've reformatted 3 times in the last month and followed the documentation for installing WSL each time. This still happens in Windows 11.

**Comment by s-h-a-d-o-w** [comment]
This can also start happening after an update.
If you stumble across this, it's probably too late already and we feel for you. 🫤

But at least now you know to _never_ upgrade WSL if you're _not_ experiencing this issue.

**Comment by widersky** [comment]
Still an issue here, very annoying :/

**Comment by aprozo** [comment]
Same here

**Comment by wuxb45** [comment]
Very annoying. Please provide an option to disable this "feature" from WSL.

**Comment by SheepReaper** [comment]
If you mount a volume formatted with NTFS, then save a file downloaded from the internet, through the wsl$ share, to a directory on the mounted NTFS volume, does it still happen?

**Comment by JacobDB** [comment]
@cianetech I think these files just contain information about where the file came from; you can delete them so long as you trust the files. I've never had an issue deleting them.

**Comment by AntonOfTheWoods** [comment]
You guys are probably all way too young. These minor inconveniences are mere trifles compared to what we had to put up with from the 90s until a year or so ago, since which we can boot to windows and we almost don't have to use it at all!

**Comment by phush0** [comment]
same here please remove it linux do not care for these files

**Comment by SheepReaper** [comment]
This issue is no longer tracked. Please go to https://github.com/microsoft/WSL/issues/7456 and upvote the OP there to gain attention.

**Comment by Octopusmode** [comment]
> I hope this gets fixed soon as it's really annoying when, like mentioned already, you copy a extracted zip file to the WSL directory

Got this bug in 2024

**Comment by PedroM25** [comment]
This is still happening to me as well

**Comment by SheepReaper** [comment]
This issue is no longer tracked. Please go to https://github.com/microsoft/WSL/issues/7456 and upvote the OP there to gain attention. @rileysjc @Octopusmode @PedroM25

**Comment by zepc007** [comment]
> This is still happening to me as well

This is still happening to me too, fix plz

**Comment by adinamarca** [comment]
Still happening.

Answer by @Octopusmode worked!

> > I hope this gets fixed soon as it's really annoying when, like mentioned already, you copy a extracted zip file to the WSL directory
> 
> Got this bug in 2024. `find /path/to/directory -type f -name "*.Identifier" -exec rm -f {} \;` as temporary solition

**Comment by italocjs** [comment]
Started happening this week when i run python scripts over wsl.

**Comment by wzr0108** [comment]
```
find . -name "*:Zone.Identifier" -type f -delete
```
it works for me

**Comment by italocjs** [comment]
Why has this been closed? this is a nightmare! all my scripts are getting broken because windows touch and create those files. i always have to do a recursive delete (as suggested above)

**Comment by nsanden** [comment]
Don't close...

**Comment by dkumza** [comment]
error persist

**Comment by italocjs** [comment]
![image](https://github.com/user-attachments/assets/adf87dc9-14ae-43d3-9bdf-68c5b0ec9267)

Please reopen! not fixed, and causes numerous programs and scripts to break

**Comment by SheepReaper** [comment]
> This issue is no longer tracked. Please go to https://github.com/microsoft/WSL/issues/7456 and upvote the OP there to gain attention. @rileysjc @Octopusmode @PedroM25 

Please follow and up vote the other issue.

**Comment by OliverRC** [comment]
Still happens

**Comment by NiklasBr** [comment]
This issue is no longer tracked. Please go to https://github.com/microsoft/WSL/issues/7456 and upvote the OP there to gain attention @OliverRC.

**Comment by ttodua** [comment]
(source: https://superuser.com/a/1847682/249349 )

1) in Start, search for `gpedit.msc`
2) navigate to `User Configuration > Administratives Templates > Windows Components > Attachment Manager`.
3) Double-click on `Do not preserve zone information in file attachment` and set it to `enabled` > ok

**Comment by italocjs** [comment]
Huh, so nearly 6 years and still nothing?? what a shame!...

**Comment by jacobseated** [comment]
**Just a minor risk vs benefit warning:**

I am not sure it is wise to change the configuration as suggested, because, as Microsoft's own documentation states 

> **By not preserving the zone information, Windows cannot make appropriate risks assessments. If you enable this policy setting, Windows does not mark file attachments by using their zone information.**
Source: https://support.microsoft.com/en-us/topic/information-about-the-attachment-manager-in-microsoft-windows-c48a4dcd-8de5-2af5-ee9b-cd795ae42738 

I think this also disables the mechanism in Windows, and that might not be such a good thing for users. Even users you should think are more in-the-know, including developers, will be totally clueless and do very dumb and unusual things sometimes.

You might judge the small risk is worth taking in your own case though... I am personally not 100% sure what it means, other than maybe not being shown those annoying "downloaded from the internet" warnings, which I have never in my life had any use for whatsoever anyway...

**Comment by Maralai** [comment]
Is this a safe and viable solution to resolve this?

![Image](https://github.com/user-attachments/assets/2c981130-de10-4ab7-9306-0724d39e6490)

From my ai research agent:

# Preventing Zone.Identifier Files in WSL

There are two main approaches to deal with Zone.Identifier files when copying from Windows to your WSL filesystem:

## Option 1: Disable Zone Information Preservation (Preventive)

The most effective solution is to disable the Windows feature that creates these files:

1. Press Win + R to open Run dialog
2. Type `gpedit.msc` and press Enter (requires Windows Pro edition)
3. Navigate to: User Configuration > Administrative Templates > Windows Components > Attachment Manager
4. Double-click on "Do not preserve zone information in file attachments"
5. Select "Enabled" and click OK

This should prevent Windows from creating Zone.Identifier files when transferring files to your WSL instance.

## Option 2: Delete Existing Zone.Identifier Files (Cleanup)

If you can't modify the policy or want to clean up existing files, you can run this command from your WSL terminal:

```bash
find . -name "*:Zone.Identifier" -type f -delete
```

Run this from the root directory where you want to search for and delete these files.

## What are Zone.Identifier Files?

These files are an NTFS feature used to store metadata about files downloaded from the internet. They help Windows identify the source of downloaded files for security purposes. The problem is that when copying files from Windows to WSL, these metadata files appear as separate files with colons in their names, which can break scripts and workflows since colons are not valid filename characters in Linux.

For most use cases in WSL/Linux, you can safely delete these files or prevent their creation without security concerns.

**Comment by zepc007** [comment]
> > I found this when doing my own research, then I tasked my agent to look into it. I hope this helps someone else in the future.
> 
> ![Image](https://github.com/user-attachments/assets/2c981130-de10-4ab7-9306-0724d39e6490)
> 
> From my ai research agent:
> 
> # Preventing Zone.Identifier Files in WSL
> There are two main approaches to deal with Zone.Identifier files when copying from Windows to your WSL filesystem:
> 
> ## Option 1: Disable Zone Information Preservation (Preventive)
> The most effective solution is to disable the Windows feature that creates these files:
> 
> 1. Press Win + R to open Run dialog
> 2. Type `gpedit.msc` and press Enter (requires Windows Pro edition)
> 3. Navigate to: User Configuration > Administrative Templates > Windows Components > Attachment Manager
> 4. Double-click on "Do not preserve zone information in file attachments"
> 5. Select "Enabled" and click OK
> 
> This should prevent Windows from creating Zone.Identifier files when transferring files to your WSL instance.
> 
> ## Option 2: Delete Existing Zone.Identifier Files (Cleanup)
> If you can't modify the policy or want to clean up existing files, you can run this command from your WSL terminal:
> 
> find . -name "*:Zone.Identifier" -type f -delete
> Run this from the root directory where you want to search for and delete these files.
> 
> ## What are Zone.Identifier Files?
> These files are an NTFS feature used to store metadata about files downloaded from the internet. They help Windows identify the source of downloaded files for security purposes. The problem is that when copying files from Windows to WSL, these metadata files appear as separate files with colons in their names, which can break scripts and workflows since colons are not valid filename characters in Linux.
> 
> For most use cases in WSL/Linux, you can safely delete these files or prevent their creation without security concerns.

This solution dose not work, still reproduce

**Comment by Maralai** [comment]
> > This should prevent Windows from creating Zone.Identifier files when transferring files to your WSL instance.
> > ## Option 2: Delete Existing Zone.Identifier Files (Cleanup)
> > If you can't modify the policy or want to clean up existing files, you can run this command from your WSL terminal:
> > find . -name "*:Zone.Identifier" -type f -delete
> > Run this from the root directory where you want to search for and delete these files.
> 
> This solution dose not work, still reproduce

Your're right, i've only found the command to delete them effective. The change to the registry did not work.

**Comment by robert-1e** [comment]
Due to the strenuous and extreme tension caused by this unintuitive and disgraceful miserable failure of a malformed system, my wife has left me. I compel you to right this dreadful wrong.

**Comment by valinet** [comment]
This "feature" sucks, please fix it, like, disable it completely from Windows, it is useless.

**Comment by etaloncop** [comment]
I would assume this is not an issue with WSL but rather some Microsoft antimalware or telemetry stuff baked into the OS. Either way it is unwanted feature and this is precisely the reason why avoid using Microsoft products whenever possible. 
....writing a comment on a microsoft platform :(

**Comment by SheepReaper** [comment]
> > This issue is no longer tracked. Please go to [#7456](https://github.com/microsoft/WSL/issues/7456) and upvote the OP there to gain attention. [@rileysjc](https://github.com/rileysjc) [@Octopusmode](https://github.com/Octopusmode) [@PedroM25](https://github.com/PedroM25)
> 
> Please follow and up vote the other issue.

It's a bit early for my annual bump, but to reiterate, please vent your frustrations (upvote) on the currently tracked issue instead of this one.

Also, this problem/feature is more fundamental than our use case. I don't have faith that it can be "fixed" in a way that satisfies our use case while not causing issues for everyone else.

NTFS supports alternate file streams. What would you have ext4 do when encountering AFS? I wager most would default to preserving that stream in a file rather than assuming and discarding it.

For me, I avoid the workflows that would trigger this from happening, but if I had to, I'd want a way to get ext4 or the driver, or some Linux utility to disable preserving AFS.

**Comment by Spenquatch** [comment]
I have started working a lot with WSL and started getting annoyed by these files so I whipped up a simple python script to recursively delete them from the directory of your choosing with a simple "unzone -r" or without the -r for non-recursive. You can download it with 'pip install unzone' or clone the repo if you want the source. [unzone](https://github.com/spenquatch/unzone)

**Comment by FranzGitHubien** [comment]
Issue still happening march 2026. It's a disgrace, almost makes using wsl2 unusable from file explorer. It happens also if you copy/paste files from within wsl2 filesystem....

**Comment by LukasSDC** [comment]
A solution that worked for me that was posted in issue #7456 consists of adding `*.wsl.localhost` in the trusted domain in Internet Options. I can now copy / create files in the Windows Explorer as expected.

> Control Panel -> Network and Internet -> Internet Options -> Security -> Local Intranet -> Sites -> Advanced -> add "wsl.localhost" as a site

<img width="1049" height="584" alt="Image" src="https://github.com/user-attachments/assets/ff4384fb-0eeb-4b71-aaff-8e93feeb4e0c" />

You can also do it with PowerShell (you need to log out and log in again to take effect) :
```
New-Item -Path "HKCU:\Software\Microsoft\Windows\CurrentVersion\Internet Settings\ZoneMap\Domains\wsl.localhost" -Force; New-ItemProperty -Path "HKCU:\Software\Microsoft\Windows\CurrentVersion\Internet Settings\ZoneMap\Domains\wsl.localhost" -Name "*" -Value 1 -PropertyType DWord -Force
```

I think wsl should add `*.wsl.localhost` as a trusted domain by default to prevent this issue

**Comment by SheepReaper** [comment]
No (*) just wsl.localhost

**Comment by LukasSDC** [comment]
I tried with and without the `*` and it seems to work with both.

If I don't put `*`, it just adds it by itself when I reopen Internet Options

**Comment by SheepReaper** [comment]
> I tried with and without the `*` and it seems to work with both.
> 
> If I don't put `*`, it just adds it by itself when I reopen Internet Options

You are correct, I did not notice that behavior before. Neat.

**Comment by jenom** [comment]
> A solution that worked for me that was posted in issue [#7456](https://github.com/microsoft/WSL/issues/7456) consists of adding `*.wsl.localhost` in the trusted domain in Internet Options. I can now copy / create files in the Windows Explorer as expected.
> 
> > Control Panel -> Network and Internet -> Internet Options -> Security -> Local Intranet -> Sites -> Advanced -> add "wsl.localhost" as a site

I can confirm, that this actually works. The group policy stopped working for me recently. Thanks for the solution
